### PR TITLE
[model] Add support for more darts models

### PR DESCRIPTION
**Pull request by ankke**
🔬 Background
Currently, we are supporting only the Linear Regression model from darts library.

🔮 Key changes
- added Random Forest regression model
- added support for other darts models in `DartsForecastingModel`
- extracted common logic to `DartsForecastingModel` and `DartsRegressionModel` classes

📋 Review Checklist

 - [x] I have performed a self-review of my own code.
  - [x] I have added docstring
  - [x] I have added typing
  - [x] I have added pytests

### CODE REVIEW

**Review by LeonieFreisinger — APPROVED** [review]
@ankke Great work! Thanks for your effort. I like the idea of introducing a class for a group of darts models and inheriting from this one for the specific models. 

Two more comments:
- please add a pytest for the RandomForest model
- can you remove the Random_Forest notebook from this PR? I think it's fine if we store the notebooks in a branch of the repo. I would not merge them into main though. 

Thank you!

File: tot/experiment.py [hunk 1]
```diff
@@ -52,10 +52,11 @@ def __post_init__(self):
         if hasattr(self.data, "freq") and self.data.freq is not None:
             data_params["freq"] = self.data.freq
         self.params.update({"_data_params": data_params})
+        model_name = self.params.get("model", self.model_class).__name__
```
Comment (ankke) [inline tot/experiment.py:55]: `DartsForecastingModel` is a wrapper around any `darts` model, which is passed by a `model` param. For models that do not require `model` param the `model_name` defaults to the class name

File: tot/models/models.py [hunk 2]
```diff
@@ -21,7 +21,6 @@ class Model(ABC):
     """
 
     params: dict
-    model_name: str
```
Comment (ankke) [inline tot/models/models.py:24]: Replaced `model_name` with `.__name__` attribute in `experiment.py`

File: tot/models/models_darts.py [hunk 3]
```diff
@@ -13,11 +13,13 @@
 
 # check import of implemented models and consider order of imports
 try:
+    from sklearn.ensemble import RandomForestRegressor
```
Comment (LeonieFreisinger) [inline tot/models/models_darts.py:16]: @ankke What is your opinion on importing the specific models from darts?
I think we should follow a clear way here of either importing all of them and proposing a child class or handing over the import part to the user (as in your notebook).

**Review by LeonieFreisinger — APPROVED** [review]
@ankke Amazing, thank you! This work provides a large improvement to our framework. :)

**Pull request merged**